src/httpserver.cpp

```diff
@@ -480,6 +475,14 @@ void StopHTTPServer()
         workQueue->WaitExit();
         delete workQueue;
     }
+    if (eventBase) {
+        LogPrint("http", "Waiting for HTTP event thread to exit\n");
+        // Give event loop a few seconds to exit (to send back last RPC responses), then break it
+        if (!threadHTTP.try_join_for(boost::chrono::milliseconds(2000))) {
```

Comment:
Out of interest, why do we have to wait?  Why can't we wait for all events to finish (and maybe just ignore all new incoming connections?)